### PR TITLE
Handle symbolic links in tarball installs.

### LINKED ISSUE #288: daml-assistant: Fails to handle symbolic links in tarball.
Reported by @robin-da.

The linux (and possibly mac) release tarball has symbolic links as part of the haskell binary packaging process. The assistant doesn't know how to handle this.

Todo:

- add regression test for installing from *unextracted* release tarball
- add symlink support in tarballs

### PR DESCRIPTION
**Pull request by fmota-da**
Fixes #288

### CODE REVIEW

File: daml-assistant/src/DAML/Assistant/Install.hs [hunk 1]
```diff
@@ -290,6 +291,9 @@ extractAndInstall env source =
                         liftIO $ setFileMode targetPath (Tar.fileMode info)
                     Tar.FTDirectory -> do
                         liftIO $ createDirectoryIfMissing True targetPath
+                    Tar.FTSymbolicLink bs -> do
+                        liftIO $ createDirectoryIfMissing True parentPath
```
Comment (moritzkiefer-da) [inline daml-assistant/src/DAML/Assistant/Install.hs]: I wonder if we should have a check that the symlink is relative.

**Comment by fmota-da** [comment]
> Not a great idea - this isn't going to work on Windows... Basically, avoid symlinks like the plague on Windows.

This is for linux. Symlinks are part of the haskell binary packaging logic.

File: daml-assistant/src/DAML/Assistant/Install.hs (hunk 1 above)
Comment (fmota-da) [reply daml-assistant/src/DAML/Assistant/Install.hs]: I'll add a check in a separate PR, thanks :)

**Comment by moritzkiefer-da** [comment]
> Symlinks are part of the haskell binary packaging logic.

Just to expand on that: Our whole packaging logic will have to be different for Windows anyway. The symlinks are due to dynamic linking and GHC on Windows does not support dynamic linking anyway.

**Pull request merged**